src/components/PairSelector/index.tsx

```diff
@@ -1,61 +1,104 @@
+'use client';
+
 import { observer } from 'mobx-react-lite';
 import { ArrowLeftRight } from 'lucide-react';
 import { AssetSelector, AssetSelectorValue } from '@penumbra-zone/ui/AssetSelector';
 import { Button } from '@penumbra-zone/ui/Button';
 import { useAssets } from '@/shared/state/assets';
 import { useBalances } from '@/shared/state/balances';
+import { useRouter } from 'next/navigation';
+import { PagePath } from '@/shared/pages.ts';
+import { getMetadataFromBalancesResponse } from '@penumbra-zone/getters/balances-response';
+import { Metadata } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+import { BalancesResponse } from '@penumbra-zone/protobuf/penumbra/view/v1/view_pb';
+import { AppRouterInstance } from 'next/dist/shared/lib/app-router-context.shared-runtime';
 
-export interface PairSelectorProps {
-  /** The `Metadata` or `BalancesResponse`, from which the swap should be initiated */
-  from?: AssetSelectorValue;
-  onFromChange?: (value?: AssetSelectorValue) => void;
+const isMetadata = (val: AssetSelectorValue): val is Metadata => {
+  return val.getType().typeName === 'penumbra.core.asset.v1.Metadata';
+};
+
+const isBalResponse = (val: AssetSelectorValue): val is BalancesResponse => {
+  return val.getType().typeName === 'penumbra.view.v1.BalancesResponse';
+};
+
+const handleRouting = ({
+  router,
+  primary,
+  numeraire,
+}: {
+  router: AppRouterInstance;
+  primary: AssetSelectorValue | undefined;
+  numeraire: AssetSelectorValue | undefined;
+}) => {
+  if (!primary || !numeraire) {
+    throw new Error('Url malformed');
+  }
+
+  let primarySymbol: string;
+  let numeraireSymbol: string;
 
-  /** The `Metadata` or `BalancesResponse`, to which the swap should be made */
-  to?: AssetSelectorValue;
-  onToChange?: (value?: AssetSelectorValue) => void;
+  // TODO: Create new getter in /web repo
+  if (isMetadata(primary)) {
+    primarySymbol = primary.symbol;
+  } else if (isBalResponse(primary)) {
+    primarySymbol = getMetadataFromBalancesResponse(primary).symbol;
+  } else {
+    throw new Error('unrecognized metadata for primary asset');
+  }
 
+  if (isMetadata(numeraire)) {
+    numeraireSymbol = numeraire.symbol;
+  } else if (isBalResponse(numeraire)) {
+    numeraireSymbol = getMetadataFromBalancesResponse(numeraire).symbol;
+  } else {
+    throw new Error('unrecognized metadata for numeraireSymbol asset');
+  }
+
+  router.push(`${PagePath.Trade}/${primarySymbol}/${numeraireSymbol}`);
+};
+
+export interface PairSelectorProps {
+  primary: AssetSelectorValue;
+  numeraire: AssetSelectorValue;
   dialogTitle?: string;
   disabled?: boolean;
 }
 
 export const PairSelector = observer(
```

Comment:
Perhaps a nextjs wizard may know what the heck this means

<img width="1390" alt="Screenshot 2024-10-22 at 8 55 03 PM" src="https://github.com/user-attachments/assets/86e7cd67-0b6d-4634-bbc7-3a1a48dfb3ce">

Reply:
Oh wait, fixed it